src/Meshes/instancedMesh.ts

```diff
@@ -171,6 +171,40 @@ export class InstancedMesh extends AbstractMesh {
         return this._sourceMesh.createInstance(name);
     }
 
+    /**
+     * Changes the source mesh for this instance
+     * Does not change instance buffers for the instance (assumes both the source meshes uses the same instance buffers)
+     * @see https://doc.babylonjs.com/how_to/how_to_use_instances
+     * @param newSourceMesh defines the new source mesh for the instance
+     * Returns nothing.
+     */
+    public changeSourceMesh(newSourceMesh: Mesh): void {

```

Comment:
not sure to get what is the goal of this function
This seems really specific and probably not suited to be in the framework itself (more a client code)

Reply:
The idea from my POV is that it is something the user might want, but easy to forget something that needs to happen. By putting it in the API it forces the API to test it before each release (which is a burden).

If we want to put it on the user then we might need to open up some things since this is impossible in typescript (but possible in javascript) from client code.

Reply:
yes but so far this code is far from complete and complex to maintain. I would see it in a subsequent PR

Reply:
Right! I will keep it for now just for my testing but I agree that it should be moved to another PR and not be part of this discussion.